lib/drawers/drawer.vue

```diff
@@ -148,82 +150,115 @@
         return _.get(this.$store, 'state.ui.currentDrawer');
       },
       isDrawerOpen() {
-        return !!this.name;
+        return !!this.name && this.activeDrawer;
       },
       tabType() {
         return this.name === 'publish-page' || 'publish-layout' ? 'icon-and-text' : 'text';
       },
       tabs() {
-        const errors = _.get(this.$store, 'state.validation.errors', []),
-          warnings = _.get(this.$store, 'state.validation.warnings', []),
-          openHealth = errors.length > 0 || warnings.length > 0;
-
-        if (this.name === 'contributors') {
-          return [{
-            title: 'Contributors',
-            component: 'contributors',
-            selected: true
-          },{
-            title: 'Page History',
-            component: 'page-history'
-          }];
-        } else if (this.name === 'layout-history') {
-          return [{
-            title: 'Layout History',
-            component: 'layout-history',
-            selected: true
-          }];
-        } else if (this.name === 'components') {
-          const state = _.get(this.$store, 'state'),
-            headLists = getHeadComponentLists(state),
-            invisibleLists = getInvisibleComponentLists(state);
-
-          return [{
-            title: 'Visible',
-            component: 'visible-components',
-            selected: true
-          }].concat(headLists).concat(invisibleLists);
-        } else if (this.name === 'preview') {
-          return [{
-            title: 'Preview',
-            component: 'preview',
-            selected: true
-          }];
-        } else if (this.name === 'publish-page') {
-          return [{
-            title: 'Health',
-            component: 'health',
-            selected: openHealth
-          }, {
-            title: 'Publish',
-            component: 'publish-page',
-            selected: !openHealth
-          }];
-        } else if (this.name === 'publish-layout') {
-          return [{
-            title: 'Health',
-            component: 'health',
-            selected: openHealth
-          }, {
-            title: 'Publish',
-            component: 'publish-layout',
-            selected: !openHealth
-          }];
-        }
+        const state = _.get(this.$store, 'state'),
+          errors = _.get(state, 'validation.errors', []),
+          warnings = _.get(state, 'validation.warnings', []),
+          hashTabName = _.get(state, 'url.sites'), // state.url.sites is just the second param in the url hash object
+          openHealth = errors.length > 0 || warnings.length > 0 || hashTabName === 'health';
+
+
+        let tabs = [
```

Comment:
I think that's a good idea and should be combined with the toolbarOptions in edit-toobar.vue perhaps, since those are the buttons that open the drawers that show these tabs.  Might be clearer if it was object something like:
[
  {
    drawer: {
    ...drawerProperties,
    tabs: [
      {
        ...tabProperties
      },
      {
        ...tabProperties
      }
    ]
  }
  },
    {
    drawer: {
      ...drawerProperties,
      tabs: [
        {
          ...tabProperties
        },
        {
          ...tabProperties
        }
      ]
    }
  }
]

That way if we ever need to add or remove a drawer, it could all be done in one place.

Reply:
This will add a lot of value to the project